influxdb/tests/server_tests/client_test_with_server.py

```diff
@@ -305,6 +358,65 @@ def test_write_points_DF(self):
             )
         )
 
+    def test_write_points_mixed_type(self):
+        """Test writing points with mixed type and reading them in again."""
+        self.assertIs(True,
+                      self.cli.write_points(mixed_series_dummy_points_part_1))
+        self.assertIs(True,
+                      self.cli.write_points(mixed_series_dummy_points_part_2))
+
+        client_1 = InfluxDBClient('localhost', self.influxd_inst.http_port,
+                                  'root', '', database='db', use_msgpack=False)
+        client_2 = InfluxDBClient('localhost', self.influxd_inst.http_port,
+                                  'root', '', database='db', use_msgpack=True)
+
+        rsp1 = client_1.query('SELECT * FROM cpu_load_short_mixed')
+        rsp2 = client_2.query('SELECT * FROM cpu_load_short_mixed')
+
+        self.assertEqual(
+            list(rsp1),
+            [[
+                {'value': 0.64,
+                 'time': '2009-11-10T23:00:00Z',
+                 "host": "server01",
+                 "region": "us-west"},
+                {'value': 0.65,
+                 'time': '2009-11-10T23:01:00Z',
+                 "host": "server01",
+                 "region": "us-west"},
+                {'value': 0.35,
+                 'time': '2009-11-10T23:02:00Z',
+                 "host": "server01",
+                 "region": "us-west"},
+                {'value': 1,
```

Comment:
The fact that influxdb returns 1 instead of 1.0 here is a bug. I don't think you should assert that the bug is present. A future version of influxdb could fix that bug, and you wouldn't want the tests too break then. I think you should just remove this assertion.

Reply:
The issue you mentioned was closed without a fix, so I doubt this will change soon.  It is a result of the default json marshaling in GoLang (they could probably work around it by customizing).

In contrast to you I believe it would be a good thing that the test would fail if the expected result would suddenly change (since the influx versions used to test are pinned it should not).

Reply:
> it would be a good thing that the test would fail if the expected result would suddenly change 

Yes, but the expected result in this case is `1.0`, not `1`, since the database contains a float. The problem is that the library does not return the expected result. I don't think this buggy behavior should be enforced by a test.

Reply:
Both are the same `Number` in JSON, which is the reason the issue was closed in InfluxDB.
I **did** remove the JSON result from the test - altough I still think it should be there, because it is a way to document the current and future behaviour.

Reply:
Yes, both are the same `Number` in JSON, but that does not make this less of a bug 😛 
And it is indeed a good idea to document the problematic behavior. Maybe you could add something to the doc comments ?

Reply:
I noticed I forgot to add the use_msgpack param docstring to InfluxDBClient doc, so I just pushed a commit documenting that paramter.

Is there a specific place you want me to describe the msgpack/Json difference?

Reply:
The doc string of the influxdb client seems like a good place to me.
Also, I'm not sure it's clear: I do not have commit rights on this repository. None of the persons with commit rights are active on the repository anymore.